lib/modules/asset/AssetService.ts

```diff
@@ -135,6 +135,97 @@ export class AssetService extends BaseService {
     });
   }
 
+  /**
+   * Update or Create an asset metadata
+   */
+  /**
+   * Update or Create an asset metadata
+   */
+  public async upsert(
+    engineId: string,
+    assetId: string,
+    metadata: Metadata,
+    request: KuzzleRequest
+  ): Promise<KDocument<AssetContent>> {
+    return lock(`asset:${engineId}:${assetId}`, async () => {
+      let asset = null;
+      try {
+        asset = await this.get(engineId, assetId, request);
+      } catch (error) {
+        this.app.log.error(`Asset ${assetId} not found`);
+        asset = {
+          _id: assetId,
+          _source: {
+            groups: [],
+            lastMeasuredAt: 0,
+            linkedDevices: [],
+            measures: undefined,
+            metadata: {},
+            model: "",
+            reference: "",
+          },
+        };
+      }
+
+      const upsertedPayload = await this.app.trigger<EventAssetUpdateBefore>(
+        "device-manager:asset:update:before",
+        { asset, metadata }
+      );
+
+      const _source = asset
```

Comment:
That ternary is unnecessary because asset is defined in the catch L157, so its condition will always be truthy.
`const _source = { ...asset._source, metadata: upsertedPayload.metadata }` should be enough in that case.

Reply:
It wasn't because asset can be null in case it do not exists